code/missionui/chatbox.cpp

```diff
@@ -888,66 +812,73 @@ void chatbox_add_line(const char *msg, int pid, int add_id)
 
 void chatbox_render_chat_lines()
 {
-   int started_at,player_num,count,ly,line_height;	
-	
-	started_at = Brief_start_display_index;
-	count = 0;	
-	ly = Chatbox_begin_y;
-	line_height = gr_get_font_height() + 1;
-	while((count < Chatbox_max_lines) && (count < Num_brief_chat_lines)){	
-		// determine what player this chat line came from, and set the appropriate text color
-		player_num = Brief_chat_lines[started_at][0];	   		
-		
-		// print the line out
-		gr_set_color_fast(Color_netplayer[player_num]);
-
-		// draw the first line (no indent)						
-		gr_string(Chatbox_begin_x + Brief_chat_indents[started_at],ly,&Brief_chat_lines[started_at][1],GR_RESIZE_MENU);		
-
-		// if we're in a team vs. team game, blit the player color icon
-		if(Netgame.type_flags & NG_TYPE_TEAM){
-			switch(Net_players[player_num].p_info.team){
-			case 0 :
-				// if he's a team captain
-				if(Net_players[player_num].flags & NETINFO_FLAG_TEAM_CAPTAIN){
-					if(Multi_common_icons[MICON_TEAM0_SELECT] != -1){
-						gr_set_bitmap(Multi_common_icons[MICON_TEAM0_SELECT]);
-						gr_bitmap(Chatbox_icon_x,ly-2,GR_RESIZE_MENU);
-					} 
-				}
-				// just you're average peon
-				else {
-					if(Multi_common_icons[MICON_TEAM0] != -1){
-						gr_set_bitmap(Multi_common_icons[MICON_TEAM0]);
-						gr_bitmap(Chatbox_icon_x,ly-2,GR_RESIZE_MENU);
+	int count = 0;	
+	int ly = Chatbox_begin_y;
+	int line_height = gr_get_font_height() + 1;
+
+	for (int i = Brief_start_display_index; i < static_cast<int>(Brief_chat.size()); i++) {	
+		if (count < Chatbox_max_lines) {
+			SCP_list<brief_chat>::iterator chat = Brief_chat.begin();
+			std::advance(chat, i);
+			
```

Comment:
This could really be replaced with something like:
```
auto chat = Brief_chat.begin();
std::advance(chat, Brief_start_display_index);

while (chat != Brief_chat.end() && count < Chatbox_max_lines) {
```
and then `++chat` when you increment the `count` value. You might also want to verify that `Brief_start_display_index` is a sane value just before this and exit early if it's not (same if `Brief_chat` is empty).

That should be a little easier to read and would definitely be faster.

Reply:
Sure. 

As I'm reading it, I'm thinking `Brief_start_display_index` should probably be in an Assertion. It should never be a bad value and if we get here and it is then there's a deeper underlying issue.

Reply:
Looks good, but you can remove the `if (count < Chatbox_max_lines)` too since it's covered by the same statement in the `while()`.

Reply:
Right.. duh. Done.